Automattic/wp-calypso
# Move preferences to redux

Moves Preferences data to redux, as listed in #5046 

- Introduces `preferences` redux tree
- Introduces `QueryPreferences` component
- Uses this in `post-editor`

[Nice and shiny readme](https://github.com/Automattic/wp-calypso/blob/064d1caa7effcd2e5b3607425265bcb440a172cc/client/state/preferences/README.md)


### Motivation
- We need it for offline - its in the editor
- I want to create `calypso-labs` to let users enable experimental features and need a place to store data

# Testing

Only preference moved is editor mode, so:
- while editing, change editor mode to html
- close window
- clear localstorage
- open new window, go to post editing and see if the mode is html


CC @rralian @gwwar @mtias @aduth @retrofox
